OSDOCS-13377: Documented the 4.17.17 z-stream notes (#88631)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

{op-system-first} adds the `irqbalance` package to `AArch64` architecture builds. This package distributes hardware interrupts across processors on a multiprocessor system, so that performance increases occur on low latency applications. (link:https://issues.redhat.com/browse/OCPBUGS-50531[*OCPBUGS-50531*])
////

[id="ocp-4-17-17-bug-fixes_{context}"]
==== Bug fixes

* Previously, the availability set fault domain count was hardcoded to `2`. This value works in most regions on {azure-first} because the fault domain counts are typically at least `2`, but failed in the `centraluseuap` and `eastusstg` regions. With this release, the availability set fault domain count in a region is set dynamically so that this issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-50017[*OCPBUGS-50017*])

* Previously, when installing a cluster on {gcp-first}, a cluster installation failed when an instance required IP Forwarding to be disabled. With this release, IP Forwarding is disabled for all {gcp-short} machines before cluster installation so that the cluster installation issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-49993[*OCPBUGS-49993*])

> * Previously, when installing a cluster on {aws-first} in existing subnets that were located in edge zones, such as a Local Zone or a Wavelength Zone, the `kubernetes.io/cluster/<InfraID>:shared` tag was missing in the subnet resources of the edge zone. With this release, a fix ensures that all subnets that are used in the `install-config.yaml ` configuration file have the required tag. (link:https://issues.redhat.com/browse/OCPBUGS-49975[*OCPBUGS-49975*])

Author: :heavy_check_mark:

// SME
* Previously, you could not install an cluster on {aws-short} in the `ap-southeast-5` region or other regions because the {product-title} internal registry did not support these regions. With this release, the internal registry is updated to include the following regions so that this issue no longer occurs:
+

> * ap-southeast-5

Author: :heavy_check_mark:

* ap-southeast-7
* ca-west-1
* il-central-1
* mx-central-1
+
(link:https://issues.redhat.com/browse/OCPBUGS-49695[*OCPBUGS-49695*])

> * Previously, when a pod was running on a node on which egress IPv6 is assigned, the pod was not able to communicate with the Kubernetes service in a dual-stack cluster. This resulted in traffic with the IP family, that the egressIP is not applicable to, being dropped. With this release, only the source network address translation (SNAT) for the IP family that the egress IPs applied to is deleted, eliminating the risk of traffic being dropped. (link:https://issues.redhat.com/browse/OCPBUGS-48828[*OCPBUGS-48828*])

Reviewer: ```suggestion
* Previously, when a pod was running on a node on which egress IPv6 is assigned, the pod was not able to communicate with the Kubernetes service in a dual-stack cluster. This resulted in traffic that the `egressIP` object is not applicable to being dropped. With this release, only the source network address translation (SNAT) for the IP family that the egress IP applied to is deleted, eliminating the risk of traffic being dropped. (link:https://issues.redhat.com/browse/OCPBUGS-48828[*OCPBUGS-48828*])
```

* Previously, when you attempted to use the {hcp} CLI to create a cluster in a disconnected environment, the installation command failed. An issue existed with the registry that hosts the command. With this release, a fix to the command registry means that you can use the {hcp} CLI to create a cluster in a disconnected environment. (link:https://issues.redhat.com/browse/OCPBUGS-48170[*OCPBUGS-48170*])

> * Previously, a bug prevented users from creating `NodePools` with ARM64 architecture on non-AWS or {azure-short} platforms. This bug resulted in validation errors that prevented the addition of bare-metal compute nodes and caused CEL validation blocks when creating a `NodePool`. The fix modifies the `NodePool` spec validation rules to allow ARM64 architecture on non-AWS or {azure-short} or Agent platforms by adding "self.platform.type == 'None'", so users can now create `NodePools` with ARM64 architecture specifications on non-AWS or {azure-short} bare-metal platforms, expanding functionality for end users. (link:https://issues.redhat.com/browse/OCPBUGS-46440[*OCPBUGS-46440*])

Reviewer: ```suggestion
* Previously, users could not create `NodePools` resources with ARM64 architecture on non-AWS or {azure-short} platforms. This resulted in validation errors that prevented the addition of bare-metal compute nodes and caused CEL validation blocks when creating a `NodePool` resource. The fix modifies the `NodePool` spec validation rules to allow ARM64 architecture on non-AWS or {azure-short} or Agent platforms by adding "self.platform.type == 'None'", so users can now create `NodePools` with ARM64 architecture specifications on non-AWS or {azure-short} bare-metal platforms, expanding functionality for end users. (link:https://issues.redhat.com/browse/OCPBUGS-46440[*OCPBUGS-46440*])
```

Reviewer: Idk, if it is resource or spec but I would add a noun in those missing spots.

Author: :heavy_check_mark:
